node/builder_chain.go

```diff
@@ -168,9 +168,10 @@ func ConfigFullNode(c interface{}) Option {
 
 		If(cfg.Client.UseIpfs,
 			Override(new(dtypes.ClientBlockstore), modules.IpfsClientBlockstore(ipfsMaddr, cfg.Client.IpfsOnlineMode)),
-			If(cfg.Client.IpfsUseForRetrieval,
+			// TODO Fix this when we use IPFS for retrieval here.
+			/*If(cfg.Client.IpfsUseForRetrieval,
 				Override(new(dtypes.ClientRetrievalStoreManager), modules.ClientBlockstoreRetrievalStoreManager),
-			),
+			),*/
```

Comment:
This feature is used by some users (afaik powergate uses this)

Reply:
@magik6k Have fixed this. We allow a retrieval client to write the retrieved CARv2 files to the IPFS node backed blockstore confgiured here. However, this "write retrieved CARv2 to the blockstore" is only done for this special `Client.IpfsUseForRetrieval` case.